### PR TITLE
fix for using bid.ext.bidType field to set bid type in PubMatic adapter response

### PR DESCRIPTION
**Pull request by PubMatic-OpenWrap**
This PR has the fix for issue #862 .
The changes here include using the bid.ext.bidType field to set the bid type in PubMatic adapter response. Default value is "banner".

### CODE REVIEW

**Comment by hhhjort** [comment]
Travis is failing:
--- FAIL: TestJsonSamples (0.00s)
    test_json.go:269: pubmatictest/exemplary/video.json:  typedBid[0].type "banner" does not match expected "video."
FAIL
coverage: 87.3% of statements
FAIL	github.com/prebid/prebid-server/adapters/pubmatic	0.040s

**Comment by hhhjort** [comment]
My guess is that the `video.json` test isn't specifying a bid with the proper `ext` for the bid type to be set as `video`.

File: adapters/pubmatic/pubmatic.go [hunk 1]
```diff
@@ -569,8 +569,18 @@ func getBidType(bidExt json.RawMessage) openrtb_ext.BidType {
 		if err == nil {
 			err = json.Unmarshal(extbyte, &bidExtMap)
 			if err == nil && bidExtMap[bidTypeExtKey] != nil {
-				bidTypeStr := bidExtMap[bidTypeExtKey].(string)
-				bidType = openrtb_ext.BidType(bidTypeStr)
+				bidTypeVal := int(bidExtMap[bidTypeExtKey].(float64))
```
Comment (hhhjort) [inline adapters/pubmatic/pubmatic.go]: Converting to a float and then an int mean that you are safe on the conversions if someone entered a float, as opposed to casting it directly to an int?

File: adapters/pubmatic/pubmatic.go [hunk 2]
```diff
@@ -550,30 +551,40 @@ func (a *PubmaticAdapter) MakeBids(internalRequest *openrtb.BidRequest, external
...
+			err = json.Unmarshal(extbyte, &bidExtMap)
+			if err == nil && bidExtMap[bidTypeExtKey] != nil {
+				bidTypeVal := int(bidExtMap[bidTypeExtKey].(float64))
+				switch bidTypeVal {
+				case 0:
+					bidType = openrtb_ext.BidTypeBanner
+				case 1:
+					bidType = openrtb_ext.BidTypeVideo
+				case 2:
+					bidType = openrtb_ext.BidTypeNative
```
Comment (guscarreon) [inline adapters/pubmatic/pubmatic.go]: The `parseImpressionObject(imp *openrtb.Imp, wrapExt *string, pubID *string)` function will filter out imps from `request.Imp[]` that are not `Video` nor `Banner`. Therefore, are we sure the `case:2` branch is reachable? Is `BidTypeNative` possible if we make it all the way here?.

**Comment by PubMatic-OpenWrap** [comment]
Thanks @guscarreon. I've added test cases for the new function.

**Pull request merged**